test/declaration_test.ts

```diff
@@ -264,13 +264,17 @@ class X {
 
 @anonymous
 @JS()
-abstract class MyMath {
-  external dynamic /*Promise<num>*/ randomInRange(num start, num end);
+abstract class MyMath {}
+
+@anonymous
+@JS('MyMath')
+abstract class _MyMath {
+  dynamic /*Promise<num>*/ randomInRange(num start, num end);
 }
 
 extension on MyMath {
-  Future randomInRangeAsFuture(num start, num end) {
-    return _promiseToFuture(this.randomInRange(start, end));
+  Future randomInRange(num start, num end) {
+    return _promiseToFuture((this as _MyMath).randomInRange(start, end));
```

Comment:
Write this as
```dart
final _MyMath t = this;
return _promiseToFuture(that.randomInRange(start, end));
```
Because that way dart2js -o3 mode will compile out the cast.

Reply:
Done. I got an error about having to cast the rhs, so I added 'this as _MyMath'.